docs/proposals/workflow.md

```diff
@@ -0,0 +1,124 @@
+A similar approach is implemented in Chronos for [dependent jobs](https://mesos.github.io/chronos/docs/api.html#adding-a-dependent-job).
+
+### API Object
+
+To implement _workflows_ the `Job` API should be modified:
+
+```go
+// JobSpec describes how the job execution will look like.
+
+// These are valid conditions of a job.
+const (
+	// JobComplete means the job has completed its execution.
+	JobComplete JobConditionType = "Complete"
+	// JobWaiting means the job is waiting for its parents to finish their tasks.
+	JobWaiting  JobConditionType = "Waiting"
+)
+
+type JobSpec struct {
+	...
+	// Job labels selector to detect parent jobs.
+	ParentSelector map[string]string `json:"parentSelector"`
+}
+
+```
+
+#### JobSpec
+
+A new labels selector will be added to `job.spec`. The `job.Spec.ParentSelector` is a label query over a set of jobs.
+If all selected jobs are completed the `job` will be started immediately.
+If `job.Spec.ParentSelector` is absent the `job` will be started immediately.
+If `job.Spec.ParentSelector` is pointing to a non-existing job, the job will wait indefinitely.
+Since labels are forwarded directly from `job` to pods, users should not create pods whose labels match this selector, either directly,
+via another Job, or via another controller (for example Replication controller), [see](https://github.com/kubernetes/kubernetes/issues/14961).
+
+#### JobStatus
+
+`job.Status` won't be modified, but a new job condition will be added.
+
+#### JobCondition
+
+A new constant value will be added to `JobConditionType` - `Waiting`. This will inform the job is waiting for its parents to finish execution. `Waiting` condition will be valid only for `workflow job`.
+
+### CRUD
+
+* Creating a job without a `job.Spec.Parentselector` (nil `job.Spec.Parentselector`) will simply follow the usual `job` life cycle.
+* Without a `job.Spec.ParentSelector` (i.e. nil) the job will be started immediately.
+* With a non nil `job.Spec.ParentSelector` the logic already described will be considered.
+* The validation will prevent the user to set an _empty_ `job.Spec.ParentSelector`.
+* Users may read the `job` using the usual `get`, `describe` commands already implemented for `job`. The `describe` command should display the non nil `job.Spec.ParentSelector` (if any).
+* A `job` cannot be updated. To update a `job`, user should delete and re-create it. The only exception is `job` _scaling_. A `workflow job` (i.e. with a non nil selector) can be scaled in the common way. No matter if pods are currently running or not.
```

Comment:
Since we've switched to normal label selector, I'd switch all non-nil to non-empty, as this reads better, imho ;)